src/main/java/org/mozilla/taskcluster/client/TaskClusterRequestHandler.java

```diff
@@ -165,4 +167,18 @@ public TaskClusterRequestHandler setBaseURL(String baseURL) {
         }
         return callSummary;
     }
+
+    /**
+     * Utility method to uri encode a parameter for using in url
+     */
+    protected static String uriEncode(String param) {
+        try {
+            return URLEncoder.encode(param, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            // This should not be possible since we specify UTF-8
+            e.printStackTrace();
+            System.exit(64);
```

Comment:
I trust your java skills that this is true... but it sounds right... Just sad they didn't do an `URLEncoder.encodeUTF8(string)` method that couldn't throw..